src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -51,7 +51,16 @@ public function loadUserByUsername($username)
             $user = $repository->findOneBy(array($this->property => $username));
         } else {
             if (!$repository instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('The Doctrine repository "%s" must implement UserProviderInterface.', get_class($repository)));
+                $message = 'You must either set the "property" option in the corresponding entity provider configuration ';
+                $repositoryClass = get_class($repository);
+
+                if ('Doctrine\ORM\EntityRepository' === $repositoryClass) {
```

Comment:
This is not fine for ODM projects, where the default repository class is not ``Doctrine\ORM\EntityRepository``.

The EntityUserProvider is not specific to the ORM. It works with the Doctrine Common APIs.